src/main/java/graphql/schema/idl/SchemaTypeChecker.java

```diff
@@ -78,6 +82,13 @@
 
         checkFieldsAreSensible(errors, typeRegistry);
 
+        if (enforceSchemaDirectives) {
```

Comment:
I would suggest to make it true by default, because the spec says: "Directives must only be used in the locations they are declared to belong in. In this example, a directive is defined which can be used to annotate a fragment definition:". So I would argue the spec requires it.

Reply:
I think we should make it on by default, because the spec says it is required that the directives are declared.